DataFormats/Phase2L1Taus/interface/L1HPSPFTau.h

```diff
@@ -0,0 +1,156 @@
+  const l1t::PFCandidateRefVector& stripPhotons()            const { return stripPhotons_;            }
+
+  const l1t::PFCandidateRefVector& isoAllL1PFCandidates()    const { return isoAllL1PFCandidates_;    }
+  const l1t::PFCandidateRefVector& isoChargedHadrons()       const { return isoChargedHadrons_;       }
+  const l1t::PFCandidateRefVector& isoElectrons()            const { return isoElectrons_;            }
+  const l1t::PFCandidateRefVector& isoNeutralHadrons()       const { return isoNeutralHadrons_;       }
+  const l1t::PFCandidateRefVector& isoPhotons()              const { return isoPhotons_;              }
+  const l1t::PFCandidateRefVector& isoMuons()                const { return isoMuons_;                }
+  
+  const l1t::PFCandidateRefVector& sumAllL1PFCandidates()    const { return sumAllL1PFCandidates_;    }
+  const l1t::PFCandidateRefVector& sumChargedHadrons()       const { return sumChargedHadrons_;       }
+  const l1t::PFCandidateRefVector& sumElectrons()            const { return sumElectrons_;            }
+  const l1t::PFCandidateRefVector& sumNeutralHadrons()       const { return sumNeutralHadrons_;       }
+  const l1t::PFCandidateRefVector& sumPhotons()              const { return sumPhotons_;              }
+  const l1t::PFCandidateRefVector& sumMuons()                const { return sumMuons_;                }
+
+  //const l1t::VertexRef&            primaryVertex()           const { return primaryVertex_;           }
+  const l1t::TkPrimaryVertexRef&   primaryVertex()           const { return primaryVertex_;           }
+
+  enum Kind { kUndefined, kOneProng0Pi0, kOneProng1Pi0, kThreeProng0Pi0, kThreeProng1Pi0 };
+  Kind tauType() const { return tauType_; }
+
+  const reco::Particle::LorentzVector& strip_p4() const { return strip_p4_; }
+  
+  float sumAllL1PFCandidates_pt() const { return sumAllL1PFCandidates_pt_; }
+  float signalConeSize()          const { return signalConeSize_;          }
+  float isolationConeSize()       const { return signalConeSize_;          }
+
+  float sumChargedIso()           const { return sumChargedIso_;           }
+  float sumNeutralIso()           const { return sumNeutralIso_;           }
+  float sumCombinedIso()          const { return sumCombinedIso_;          }
+  float sumChargedIsoPileup()     const { return sumChargedIsoPileup_;     }
+  float rhoCorr()                 const { return rhoCorr_;                 }
+  
+  bool passTightIso()     const { return passTightIso_;     }
+  bool passMediumIso()    const { return passMediumIso_;    }
+  bool passLooseIso()     const { return passLooseIso_;     }
+  bool passVLooseIso()    const { return passVLooseIso_;    }
+  
+  bool passTightRelIso()  const { return passTightRelIso_;  }
+  bool passMediumRelIso() const { return passMediumRelIso_; }
+  bool passLooseRelIso()  const { return passLooseRelIso_;  }
+  bool passVLooseRelIso() const { return passVLooseRelIso_; }
+  
+  friend class ::L1HPSPFTauBuilder;
```

Comment:
friend classes are usually not a great idea. Can we just make accessors.

Reply:
On the other hand accessors are available to everybody.

If using the individual accessors on a non-`const` object would break encapsulation (*e.g.* break object invariants) then having accessors is just like having public members.

The usual approach would be to have the constructor responsible for setting all the data, so the object always exists in a valid state.

An other approach would be to have only one (or a few) methods to "fill" the whole object, so that the object is either "empty" (default constructed) or valid.